Code/Framework/AzFramework/Platform/Common/Xcb/AzFramework/XcbNativeWindow.cpp

```diff
@@ -219,17 +231,143 @@ namespace AzFramework
         }
     }
 
+
+    // This function will draw the splash image if one is set
+    // otherwise it will return without doing anything
+    void XcbNativeWindow::DrawSplash()
+    {
+#ifndef SPLASH_IMAGE_EXISTS
+    return;  // failed to find splash image no need to draw
+#else
+        const unsigned int bytesPerPixel = 4;
+        
+        typedef struct
+        {
+            const png_byte* data;
+            const png_size_t size;
+        } DataHandle;
```

Comment:
No need for `typedef struct ..`. This is a c++ source file, you can declare this using c++ syntax.